components/brave_wallet_ui/components/extension/confirm-transaction-panel/index.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from 'react'
+import { create } from 'ethereum-blockies'
+import { WalletAccountType, Network, TransactionPanelPayload } from '../../../constants/types'
+import { reduceAddress } from '../../../utils/reduce-address'
+import { NetworkOptions } from '../../../options/network-options'
+import locale from '../../../constants/locale'
+import { formatBalance, formatFiatBalance } from '../../../utils/format-balances'
+import { NavButton } from '../'
+
+// Styled Components
+import {
+  StyledWrapper,
+  FromCircle,
+  ToCircle,
+  AccountNameText,
+  TopRow,
+  NetworkText,
+  TransactionAmmountBig,
+  TransactionFiatAmountBig,
+  GrandTotalText,
+  MessageBox,
+  TransactionTitle,
+  TransactionTypeText,
+  TransactionText,
+  ButtonRow,
+  MoreButton,
+  AccountCircleWrapper,
+  ArrowIcon,
+  FromToRow,
+  Divider,
+  SectionRow,
+  SectionRightColumn,
+  EditButton
+} from './style'
+
+export interface Props {
+  selectedAccount: WalletAccountType
+  transactionPayload: TransactionPanelPayload
+  selectedNetwork: Network
+  onConfirm: () => void
+  onReject: () => void
+  onClickMore: () => void
+}
+
+function ConfirmTransactionPanel (props: Props) {
+  const {
+    selectedAccount,
+    selectedNetwork,
+    transactionPayload,
+    onConfirm,
+    onReject,
+    onClickMore
+  } = props
+
+  const fromOrb = React.useMemo(() => {
+    return create({ seed: selectedAccount.address, size: 8, scale: 16 }).toDataURL()
+  }, [selectedAccount.address])
+
+  const toOrb = React.useMemo(() => {
+    return create({ seed: transactionPayload.toAddress, size: 8, scale: 10 }).toDataURL()
+  }, [transactionPayload])
+
+  const formatedAmounts = React.useMemo(() => {
+    const sendAmount = formatBalance(transactionPayload.transactionAmount, transactionPayload.erc20Token.decimals)
+    const sendFiatAmount = formatFiatBalance(transactionPayload.transactionAmount, transactionPayload.erc20Token.decimals, transactionPayload.tokenPrice)
+    const gasAmount = formatBalance(transactionPayload.transactionGas, 18)
+    const gasFiatAmount = formatFiatBalance(transactionPayload.transactionGas, 18, transactionPayload.ethPrice)
```

Comment:
Two remarks:
* Can pricing information be separated from transaction payload?
* Are `tokenPrice` and `ethprice` static in the `transactionPayload`?

Reply:
I could separate them out, the reason I have everything together here is to make one simple return from the reducer once that is set up, and we would be able to grab the `tokenPrice` and `ethPrice` during that time.

Reply:
Discussed internally that we should probably have some reusable hook that provides live pricing information. Let's keep that for later.